src/test/parse-fail/issue-33569.rs

```diff
@@ -0,0 +1,15 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// compile-flags: -Z no-analysis
+
+macro_rules! foo {
+  { $+ } => () //~ ERROR: expected identifier, found `+`
+}              //~^ ERROR: no rules expected the token `}`
```

Comment:
You're right about the error message.

I did it this way mostly because it matches the behaviour of stable rust (1.8.0).

But I'm happy to change it. I have no strong feelings about which is better. Ideally I think the second error message should not be emitted at all, but that would probably be a bigger change (which I haven't looked into).